src/components/UrlUtilsService.js

```diff
@@ -48,13 +48,44 @@ goog.provide('ga_urlutils_service');
                   /.*kmz$/.test(url));
         };
 
-        this.proxifyUrl = function(url) {
+        // Test using a head request if the remote resource enables CORS
+        this.isCorsEnabled = function(url) {
+          var deferred = $q.defer();
+          $http.head(url, { timeout: 1500 }).then(function(response) {
+            // If the head request responded successfully it means that CORS
+            // is enabled
+            deferred.resolve(true);
+          }, function() {
+            deferred.resolve(false);
+          });
+          return deferred.promise;
+        };
+
+        this.proxifyUrlInstant = function(url) {
           if (this.needsProxy(url)) {
             return gaGlobalOptions.ogcproxyUrl + encodeURIComponent(url);
           }
           return url;
         };
 
+        this.proxifyUrl = function(url) {
```

Comment:
This is not a duplicate we have `&&` operator instead of `||` in order to test CORS, we must statisfy all these conditions.

I can make one function with a sync param but honeslty, I don't see the point.